src/arch/arm/machine/hardware.c

```diff
@@ -31,8 +31,12 @@ BOOT_CODE const p_region_t *get_avail_p_regs(void)
 
 BOOT_CODE void map_kernel_devices(void)
 {
-    for (unsigned int i = 0; i < ARRAY_SIZE(kernel_devices); i++) {
-        const kernel_frame_t *frame = &kernel_device_regions[i];
+    /* If there are no kernel frames at all, then kernel_frames is NULL. Thus we
+     * can't use ARRAY_SIZE(kernel_frames) here directly, but have to use
+     * NUM_KERNEL_FRAMES that is defined accordingly.
+     */
+    for (unsigned int i = 0; i < NUM_KERNEL_FRAMES; i++) {
+        const kernel_frame_t *frame = &kernel_frames[i];
```

Comment:
I don't understand why this is being renamed. We aren't mapping generic frames, we're mapping devices, in a function called `map_kernel_devices`. Seems like more needless churn. 

Also, this commit does a lot more than is said in the commit message. The part it doesn't admit to doing is the useful part of the commit that should be kept.

Reply:
Thanks for the feedback, I've split up the commit further also the clarified renaming. This was a misunderstanding on my side in the first place. I have picked the name `kernel_device_frame` now because this seems to be what the structure contains. A device can use more than one page, so this structure can have more elements than kernel devices and this must not be mixed up. I've see the case on RISC-V, where a peripheral has 48 KiByte and there are 12 entries here. Not sure is there is a similar case on an ARM platform,